### PR TITLE
Use caret for dependency ranges

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Clean up dependencies to resolve long resolution times on when installing `griptape[all]`
## Issue ticket number and link
https://discord.com/channels/1096466116672487547/1291202108233748492

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -224,6 +212,7 @@ pytest-cov = "^5.0.0"
 pytest-env = "^1.1.1"
 fuzzywuzzy = "^0.18.0"
 pytest-clarity = "^1.0.1"
+torch = "^2.4.1"
```
Comment (collindutter) [inline pyproject.toml:215]: Kind of clunky moving this to `test` deps but that's the only time we require it.

**Review by vachillo — DISMISSED** [review]
🥕
are you able to update dependabot to use carets?

**Comment by collindutter** [comment]
@vachillo just updated dependabots versioning strategy

**Review by vachillo — APPROVED** [review]
🥕

**Pull request merged**